### PR TITLE
CANS-932. Improve performance for Read Staff Latest Assessments operation

### PR DESCRIPTION
**Pull request by denys-davydov**
## JIRA
https://osi-cwds.atlassian.net/browse/CANS-932

## Description
The main performance improvement work for this task is made on the backend within the following PR https://github.com/ca-cwds/cans-api/pull/201 
With this PR the frontend processing of the assessments (like filtering by status and sorting by timestamps) is removed, the data is prepared on the backend fully.

## Tests

- [ ] I used TDD to develop the feature
- [ ] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I ran all my tests locally which were green.
- [ ] My code adds no new issues to Code Climate
- [ ] I ran all the linters for the project.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@
     resources :staff, only: [:show] do
       collection do
         get 'subordinates', to: 'staff#subordinates_index'
-        get 'assessments', to: 'staff#assessments'
+        get 'assessments/latest', to: 'staff#assessments'
```
Comment (wmitchell-cambria) [inline config/routes.rb]: I think... unless we expose another way to fetch assessments for a staffperson, we could leave this Rails endpoint at `'assessments'`, and only change the `StaffRepository`. That would avoid the 'conversion' from `assessments/latest` (Rails endpoint and route) to `assessments` (Rails controller method) and back to `assessments/latest` (CANS API endpoint).

Alternatively, we could name it `assessments/latest` or `assessments_latest` all the way through. (I think there is a better naming scheme for this, but I am not clear on it.)

File: app/javascript/Application/components/Search/AssessmentHistory/SearchAssessmentHistory.js [hunk 2]
```diff
@@ -2,62 +2,34 @@ import React from 'react'
...
-
-  renderAssessments() {
-    const { loadingState, navFrom, inheritUrl } = this.props
-    const assessments = this.processAssessments()
-    const noDataText =
-      loadingState === LoadingState.ready ? 'No assessments currently exist for the clients.' : 'Loading assessments...'
-
-    return assessments.length === 0 ? (
-      <div id="no-data">{noDataText}</div>
+const SearchAssessmentHistory = props => (
```
Comment (wmitchell-cambria) [inline app/javascript/Application/components/Search/AssessmentHistory/SearchAssessmentHistory.js]: I suspect now that this is simplified a bit, that we can share much more code between here and ClientAssessmentHistory. I'll look into it a bit and create a tech debt story if it looks worthwhile.

Comment (denys-davydov) [reply app/javascript/Application/components/Search/AssessmentHistory/SearchAssessmentHistory.js]: The sorting function that was previously used here is still needed for at least one other component, so it should stay for now.

**Comment by wmitchell-cambria** [comment]
Removing the blocker, per comments on the backend PR. Note that Jenkins is failing due to a code coverage threshold. We probably need to add back a few tests.

File: app/javascript/Application/components/Search/AssessmentHistory/SearchAssessmentHistory.js (hunk 2 above)
Comment (denys-davydov) [reply app/javascript/Application/components/Search/AssessmentHistory/SearchAssessmentHistory.js]: @wmitchell-cambria  And I will refactor the function so the tech debt story will not be needed.

**Comment by wmitchell-cambria** [comment]
Refactoring changes look OK.

**Pull request merged**